src/attributes.rst

```diff
@@ -1745,8 +1745,30 @@ Attribute ``panic_handler``
 .. rubric:: Legality Rules
 
 :dp:`fls_ryz8qy1wdnma`
-:t:`Attribute` :c:`panic_handler` shall apply to :t:`[function]s` with
-:t:`function signature` ``fn(&core::panic::PanicInfo) -> !``.
+:t:`Attribute` :c:`panic_handler` shall apply to :t:`[function]s` with the
+following restrictions:
+
+* :dp:`fls_ncyod5gegMqL`
+  It lacks :t:`function qualifiers`,
+
+* :dp:`fls_jzIHzPkJqVm8`
+  It's :t:`ABI` is Rust,
+
+* :dp:`fls_qXii07xWRUoT`
+  It lacks :t:`generic parameters`,
+
+* :dp:`fls_ZEjvn4mmfSct`
+  It has a single :t:`function parameter` with the type being a :t:`shared
+  reference type` whose inner :t:`type` is :std:`core::panic::PanicInfo`,
```

Comment:
It might be cleaner to just state it as

It has a single function parameter whose type is & core::panic::PanicInfo,